src/components/client/ConfirmCourtReservation.js

```diff
@@ -17,212 +17,220 @@ class ConfirmCourtReservation extends Component {
   state = { selectedIndex: 0, priceButtons: [], prices: [] };
 
   static navigationOptions = ({ navigation }) => {
-      return {
-          headerLeft: navigation.getParam('leftButton')
-      };
+    return {
+      headerLeft: navigation.getParam('leftButton')
+    };
   };
 
-  async componentDidMount() {
-      this.props.navigation.setParams({
-          leftButton: this.renderBackButton()
-      });
+  componentDidMount() {
+    this.props.navigation.setParams({
+      leftButton: this.renderBackButton()
+    });
 
-      await this.priceButtons();
-      this.onPriceSelect(0);
+    this.priceButtons();
   }
 
   renderBackButton = () => {
-      return <HeaderBackButton onPress={this.onBackPress} tintColor='white' />
-  }
+    return <HeaderBackButton onPress={this.onBackPress} tintColor="white" />;
+  };
 
   onBackPress = () => {
-      this.onNewReservation();
-      this.props.navigation.goBack(null);
-  }
+    this.onNewReservation();
+    this.props.navigation.goBack(null);
+  };
 
   priceButtons = () => {
-      const { court } = this.props;
-      const priceButtons = [];
-      const prices = [];
-
-      if (court) {
-          priceButtons.push(`Sin Luz: $${court.price}`);
-          prices.push(court.price);
-
-          if (court.lightPrice) {
-              priceButtons.push(`Con Luz: $${court.lightPrice}`);
-              prices.push(court.lightPrice);
-          }
+    const { court } = this.props;
+    const priceButtons = [];
+    const prices = [];
+
+    if (court) {
+      priceButtons.push(`Sin Luz: $${court.price}`);
+      prices.push(court.price);
+
+      if (court.lightPrice) {
+        priceButtons.push(`Con Luz: $${court.lightPrice}`);
+        prices.push(court.lightPrice);
       }
+    }
 
-      this.setState({ priceButtons, prices });
-  }
+    this.setState({ priceButtons, prices }, this.onPriceSelect(0));
```

Comment:
this.onPriceSelect(0) en realidad deberia ser llamado con arrow function () => this.onPriceSelect(0) para que no se ejecute apenas se llega a esa linea sino despues del setState